grails-app/services/org/pih/warehouse/order/OrderService.groovy

```diff
@@ -168,6 +168,39 @@ class OrderService {
                         ilike("description", "%" + params.q + "%")
                         ilike("orderNumber", "%" + params.q + "%")
                     }
+
+                    orderItems {
+                        product {
+                            or {
+                                ilike("name", "%" + params.q + "%")
+                                ilike("productCode", "%" + params.q + "%")
+                            }
+                        }
+                    }
+
+                    orderItems {
+                        inventoryItem {
+                            ilike("lotNumber", "%" + params.q + "%")
+                        }
+                    }
+
+                    orderItems {
+                        originBinLocation {
+                            or {
+                                ilike("name", "%" + params.q + "%")
+                                ilike("locationNumber", "%" + params.q + "%")
+                            }
+                        }
+                    }
+
+                    orderItems {
+                        destinationBinLocation {
+                            or {
+                                ilike("name", "%" + params.q + "%")
+                                ilike("locationNumber", "%" + params.q + "%")
+                            }
+                        }
+                    }
```

Comment:
To the person who will be doing the review: Please make sure that this not affect existing filtering.
In my understainding it onlly adds new options.

Reply:
I think it is ok, but would need to be tested